sawmill-core/src/test/java/io/logz/sawmill/PipelineTest.java

```diff
@@ -148,4 +163,34 @@ public void testConditionalElse() {
         ProcessorExecutionStep onFalseExecutionStep = (ProcessorExecutionStep) conditionalExecutionStep.getOnFalse().get(0);
         assertThat(onFalseExecutionStep.getProcessor()).isInstanceOf(AddTagProcessor.class);
     }
+
+    @Test(expected = SawmillException.class)
+    public void shouldFailPipelineCreationOnNoGeoIpConfigurationGiven() throws Exception {
+        new Pipeline.Factory().create("test", new PipelineDefinition(
+                singletonList(new ProcessorExecutionStepDefinition(new ProcessorDefinition(
+                        "geoIp", new HashMap<>()), null, null, null
+                )),
+                true
+        ));
+    }
+
+    @Test(expected = SawmillException.class)
+    public void shouldFailPipelineCreationOnGeoIpConfigurationInvalidFileGiven() throws Exception {
+        new Pipeline.Factory(new GeoIpConfiguration("LICENSE")).create("test", new PipelineDefinition(
+                singletonList(new ProcessorExecutionStepDefinition(new ProcessorDefinition(
+                        "geoIp", new HashMap<>()), null, null, null
+                )),
+                true
+        ));
+    }
+
+    @Test(expected = SawmillException.class)
+    public void shouldFailPipelineCreationOnGeoIpConfigurationFileDoesNotExist() throws Exception {
+        new Pipeline.Factory(new GeoIpConfiguration("non-existing-file")).create("test", new PipelineDefinition(
+                singletonList(new ProcessorExecutionStepDefinition(new ProcessorDefinition(
+                        "geoIp", new HashMap<>()), null, null, null
+                )),
+                true
+        ));
+    }
```

Comment:
Maybe we want in these tests to verify the error message (i.e. "geo ip database configuration is not provided", "... configuration file not exists", "... configuration file is corrupt", etc.)

Reply:
BTW as your comment, we can consider move this tests (and a successful one) to the `GeoIpProcessorTest`, or create a new one `GeoIpProcessorIntegrationTest`. WDYT?